### PR TITLE
portal: Add meson support

### PR DESCRIPTION
**Pull request by A6GibKm**
# Test on Silverblue

Temporarily disable selinux

    setenforce 0

Create /run/extensions

    mkdir /run/extensions

Have a preferences file `~/.config/xdg-desktop-portal/gnome-portals.conf` with contents

```
[preferred]
default=gnome;gtk;
org.freedesktop.impl.portal.Secret=oo7-portal;
```

and restart xdg-desktop-portal

```
systemctl --user restart xdg-desktop-portal
```

From oo7/portal run in a toolbox:

    meson --prefix=/usr _build -Dprofile=development
    ninja -C _build && DESTDIR=oo7-extension meson install -C _build

From outside the sandbox:

```
cp -r _build/oo7-extension /run/extensions/ && systemd-sysext refresh --force
```

One can get more debug output by replacing services for xdp and oo7-portals:
```
/usr/libexec/xdg-desktop-portal -r -v
```

```
RUST_LOG=oo7_portal=debug RUST_BACKTRACE=full /usr/libexec/oo7-portal
```

### CODE REVIEW

File: portal/Cargo.toml [hunk 1]
```diff
@@ -14,12 +14,12 @@ version.workspace = true
 [dependencies]
 futures-channel.workspace = true
 futures-util.workspace = true
-oo7 = { workspace = true, features = ["unstable"] }
+oo7 = { workspace = true, default-features = false, features = ["tokio", "native_crypto", "unstable"] }
```
Comment (bilelmoussaoui) [inline portal/Cargo.toml:17]:
```suggestion
oo7 = { workspace = true, features = ["tokio", "native_crypto", "unstable"] }
```
you set that in the workspace already, also it is weird that the cli uses native crypto & async-std and the portal uses native / tokio, which are the default-features anyway

File: portal/src/main.rs [hunk 2]
```diff
@@ -139,6 +141,8 @@ async fn main() -> Result<(), zbus::Error> {
         | zbus::fdo::RequestNameFlags::AllowReplacement;
     cnx.request_name_with_flags(PORTAL_NAME, flags).await?;
 
+    tracing::info!("Got dbus name {PORTAL_NAME}");
```
Comment (bilelmoussaoui) [inline portal/src/main.rs:144]: what is this for ?

File: portal/src/meson.build [hunk 3]
```diff
@@ -0,0 +1,30 @@
+cargo = find_program('cargo', required: true)
```
Comment (bilelmoussaoui) [inline portal/src/meson.build:1]:
```suggestion
cargo = find_program('cargo')
```

File: portal/src/meson.build [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+is_devel = true
```
Comment (bilelmoussaoui) [inline portal/src/meson.build:3]: should this stay true?

File: portal/src/main.rs (hunk 2 above)
Comment (A6GibKm) [reply portal/src/main.rs:144]: Just to know that the binary reached the "main" loop without erring out, but it might be unnecessary.

Comment (A6GibKm) [reply portal/src/main.rs]: I removed it, if the process fails it will just close.

File: portal/Cargo.toml (hunk 1 above)
Comment (A6GibKm) [reply portal/Cargo.toml]: I removed the redundant default-features.

File: portal/src/main.rs [hunk 5]
```diff
@@ -125,18 +126,33 @@ async fn send_secret_to_app(app_id: &str, fd: zvariant::OwnedFd) -> Result<(), E
...
+    replace: bool,
+}
+
 #[tokio::main]
 async fn main() -> Result<(), zbus::Error> {
+    let args = Args::parse();
+
     tracing_subscriber::fmt::init();
 
+    tracing::info!("Initializing oo7-portal {}", env!("CARGO_PKG_VERSION"));
```
Comment (bilelmoussaoui) [inline portal/src/main.rs:144]:
```suggestion
    tracing::info!("Initializing {} {}", env!("CARGO_BIN_NAME"), env!("CARGO_PKG_VERSION"));
```

**Pull request merged**